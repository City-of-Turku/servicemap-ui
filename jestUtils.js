import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });