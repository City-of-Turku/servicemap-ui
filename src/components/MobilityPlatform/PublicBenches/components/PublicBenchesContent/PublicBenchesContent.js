import PropTypes from 'prop-types';
import React from 'react';
import { Typography } from '@mui/material';

const PublicBenchesContent = ({ classes, intl, item }) => {
  const singleValText = (messageId, value) => (
    <div className={classes.margin}>
      <Typography component="p" variant="body2">
        {intl.formatMessage({ id: messageId }, { value })}
      </Typography>
    </div>
  );

  const publicBenchInfo = (
    <div className={classes.container}>
      <div className={classes.headerContainer}>
        <Typography variant="subtitle1" component="h3">
          {intl.formatMessage({ id: 'mobilityPlatform.content.publicBench.title' })}
        </Typography>
      </div>
      <div className={classes.textContainer}>
        {singleValText('mobilityPlatform.content.publicBench.condition', item.extra.Kunto)}
        {singleValText('mobilityPlatform.content.publicBench.model', item.extra.Malli)}
      </div>
    </div>
  );

  return (
    <div className={classes.padding}>
      {publicBenchInfo}
    </div>
  );
};

PublicBenchesContent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.shape({
    formatMessage: PropTypes.func,
  }).isRequired,
  item: PropTypes.objectOf(PropTypes.any),
};

PublicBenchesContent.defaultProps = {
  item: {},
};

export default PublicBenchesContent;
