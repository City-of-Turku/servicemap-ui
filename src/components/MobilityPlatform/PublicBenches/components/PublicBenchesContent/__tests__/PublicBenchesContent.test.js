// Link.react.test.js
import React from 'react';
import PublicBenchesContent from '../index';
import finnishTranslations from '../../../../../../i18n/fi';
import { getRenderWithProviders } from '../../../../../../../jestUtils';

const mockProps = {
  item: {
    extra: {
      Malli: 'Testipenkki',
      Kunto: 'Hyv√§',
    },
  },
};

const renderWithProviders = getRenderWithProviders();

describe('<PublicBenchesContent />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<PublicBenchesContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<PublicBenchesContent {...mockProps} />);
    const p = container.querySelectorAll('p');
    const h3 = container.querySelectorAll('h3');
    expect(h3[0].textContent).toContain(finnishTranslations['mobilityPlatform.content.publicBench.title']);
    expect(p[0].textContent).toContain(`Penkin kunto: ${mockProps.item.extra.Kunto}`);
    expect(p[1].textContent).toContain(`Penkin malli: ${mockProps.item.extra.Malli}`);
  });
});
