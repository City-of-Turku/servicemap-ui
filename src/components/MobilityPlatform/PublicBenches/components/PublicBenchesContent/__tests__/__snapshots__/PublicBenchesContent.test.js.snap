// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublicBenchesContent /> should match snapshot 1`] = `
<div>
  <div
    class="injectIntl(PublicBenchesContent)-padding-1"
  >
    <div>
      <div
        class="injectIntl(PublicBenchesContent)-headerContainer-2"
      >
        <h3
          class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
        >
          Yleinen penkki
        </h3>
      </div>
      <div
        class="injectIntl(PublicBenchesContent)-textContainer-3"
      >
        <div
          class="injectIntl(PublicBenchesContent)-margin-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            Penkin kunto: Hyv√§
          </p>
        </div>
        <div
          class="injectIntl(PublicBenchesContent)-margin-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            Penkin malli: Testipenkki
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
